src/fs/tlog.c

```diff
@@ -625,7 +625,10 @@ boolean log_write_eav(log tl, tuple e, symbol a, value v)
     u64 len = buffer_length(tl->tuple_staging);
     if ((tl->state == TLOG_STATE_FAILED) || len >= TFS_LOG_MAX_TUPLE_STAGING_BYTES)
         return false;
+    tuple parent = (v && is_tuple(v)) ? cleanup_directory(v) : 0;
```

Comment:
Although not likely to be an issue today - given that we're currently only writing non-filesystem metadata at once during mkfs, and not updating it during runtime (except for writing the booted flag) - it seems that moving the directory cleanup and fixup to this level might interfere with updating a non-tfs tuple that happens to contain a ".." attribute.

Reply:
OK, I moved the logic to clean up and fix up a directory tree to the filesystem_write_eav() function, which now takes an additional boolean argument indicating whether a cleanup (and subsequent fixup) should be performed